llmfoundry/data/finetuning/tasks.py

```diff
@@ -1033,7 +1034,16 @@ def build_from_streaming(
         *args: Any,
         **kwargs: Any,
     ) -> StreamingFinetuningDataset:
-        return self.streaming_dataset_class(*args, **kwargs)
+        dataset = self.streaming_dataset_class(*args, **kwargs)
+        num_canonical_nodes = dataset.num_canonical_nodes
+        num_samples = dataset.num_samples
+        if num_canonical_nodes is not None and num_canonical_nodes < num_samples:
+            raise DatasetTooSmallError(
```

Comment:
Is this something that's actionable by the user? isn't the index.json not accessible by them?

Reply:
We would tell them in the UI that they need to read docs and make sure they specify enough data. See [here for exact message.](https://github.com/databricks-mosaic/mcloud/blob/0d7a5066a7f30511fd4d31c2e8c1ed57fe72605a/mapi/src/graphql/runEvent/runEventCustomExceptions.ts#L174) Would this solve the issue?


The exception messages we write here are just for raw foundry users.

Reply:
Yeah I think we would just ask them to add more data for now. Ideally we should calculate data minimums across all models to give a good solid token count that shouldn't fail.